scipy/stats/qmc.py

```diff
@@ -0,0 +1,647 @@
+"""Quasi-Monte Carlo methods.
+
+Define function to generate sample of points in the unit hypercube.
+
+"""
+
+from __future__ import division
+
+import copy
+import numpy as np
+from scipy.optimize import brute
+from scipy._lib._util import check_random_state
+from scipy.optimize import basinhopping
+
+__all__ = ['discrepancy', 'halton', 'orthogonal_latin_hypercube',
+           'latin_hypercube', 'optimal_design']
+
+
+def discrepancy(sample, bounds=None, iterative=False):
+    """Centered discrepancy on a given sample..
+
+    The discrepancy is a uniformity criterion used to assess the space filling
+    of a number of samples in a hypercube.
+    The discrepancy measures how the spread of the points deviates from a
+    uniform distribution.
+    The lower the value is, the better the coverage of the parameter space is.
+
+    Parameters
+    ----------
+    sample : array_like (n_samples, k_vars)
+        The sample to compute the discrepancy from.
+    bounds : tuple or array_like ([min, k_vars], [max, k_vars])
+        Desired range of transformed data. The transformation apply the bounds
+        on the sample and not the theoretical space, unit cube. Thus min and
+        max values of the sample will coincide with the bounds.
+    iterative : bool
+        Must be False if not using it for updating the discrepancy.
+
+    Returns
+    -------
+    discrepancy : float
+        Centered discrepancy.
+
+    References
+    ----------
+    [1] Fang et al. "Design and modeling for computer experiments",
+      Computer Science and Data Analysis Series Science and Data Analysis
```

Comment:
Looks up the reference, I see "Computer Science and Data Analysis Series" without the stuff that comes after, which reads a bit like duplication?